### ISSUE TITLE
Model save issue

### ISSUE THREAD

**Issue by mhyeonsoo** [4uiiurz1/keras-arcface#17]
Hello, thanks for a great codes.

I am now implementing Res50 backbone arcface metric on it, and have been facing trouble in saving model file.

The error being shown up is, 
 Layers with arguments in `__init__` must override `get_config`. 

which may require configuration overriding of the model and I am not pretty sure about it.

I referred vgg_arcface function to make Res50_arcface function and below it the result.

--------------------------------------------------------------------------------
    
    # Resnet Backbone
    def ResNet50_arcface(args):
        y = Input(shape=(200,))
        x = inputs = Input([224, 224, 3], name='input_image')
        x = ResNet50(input_shape=x.shape[1:], include_top=False, weights='imagenet')(x)

        # BN + Dropout + FC + BN ( as paper mentioned )
        x = BatchNormalization()(x)
        x = Dropout(rate=0.5)(x)
        x = Flatten()(x)
        x = Dense(args.num_features, kernel_regularizer=regularizers.l2(5e-4))(x)
        x = BatchNormalization()(x)
    
        output = ArcFace(200, regularizer=regularizers.l2(weight_decay))([x, y])
  
        return Model([inputs, y], output)
--------------------------------------------------------------------------------

Could anyone give me suggestion?

Thank you

**Comment by IgorSondors** [comment]
> Hello, thanks for a great codes.
> 
> I am now implementing Res50 backbone arcface metric on it, and have been facing trouble in saving model file.
> 
> The error being shown up is, Layers with arguments in `__init__` must override `get_config`.
> 
> which may require configuration overriding of the model and I am not pretty sure about it.
> 
> I referred vgg_arcface function to make Res50_arcface function and below it the result.
> 
> ```
> # Resnet Backbone
> def ResNet50_arcface(args):
>     y = Input(shape=(200,))
>     x = inputs = Input([224, 224, 3], name='input_image')
>     x = ResNet50(input_shape=x.shape[1:], include_top=False, weights='imagenet')(x)
> 
>     # BN + Dropout + FC + BN ( as paper mentioned )
>     x = BatchNormalization()(x)
>     x = Dropout(rate=0.5)(x)
>     x = Flatten()(x)
>     x = Dense(args.num_features, kernel_regularizer=regularizers.l2(5e-4))(x)
>     x = BatchNormalization()(x)
> 
>     output = ArcFace(200, regularizer=regularizers.l2(weight_decay))([x, y])
> 
>     return Model([inputs, y], output)
> ```
> 
> Could anyone give me suggestion?
> 
> Thank you

How did you solve the problem?
